scripts/factory-agent.sh

```
@@ -710,6 +723,53 @@ copy_worktree_env() {
+# roots (repo root + apps/* + packages/*) are seeded — never the factory's own
+# worktrees/ checkouts.
+seed_worktree_node_modules() {
+  local wt="$1" src rel
+  for src in "$REPO_ROOT"/node_modules "$REPO_ROOT"/apps/*/node_modules "$REPO_ROOT"/packages/*/node_modules; do
+    [[ -d "$src" ]] || continue
+    rel="${src#"$REPO_ROOT"/}"
+    [[ -e "$wt/$rel" ]] && continue
+    mkdir -p "$wt/$(dirname "$rel")"
+    if ! cp -c -R "$src" "$wt/$rel" 2>>"$LOG_FILE"; then
+      log "WARNING: clonefile seed of $rel failed; pnpm install will repopulate it"
+      rm -rf "$wt/$rel" 2>/dev/null || true
```

Shellcheck (0.11.0)

[warning] Use "${var:?}" to ensure this never expands to / .

(SC2115)